File: src/main/java/cccs/oldversion/RulesConf.java
package cccs.oldversion;

import java.util.List;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;

public class RulesConf {
	public List<RuleConf> rules;

	public static class TagConf {
		public String name;
	}

	public static class RuleConf {
		public String rulename;
		public String description;
		public String action;
		public List<TagConf> tags;
		public List<String> groupby;
		public boolean ordered;
		public String parent;
		public String child;
	}

	public static RulesConf load(String spec) throws Exception {
        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
        objectMapper.findAndRegisterModules();
        return objectMapper.readValue(spec, RulesConf.class);
    }
}

File: src/main/scala/cccs/fluxcapacitor/RulesConf.scala
package cccs.fluxcapacitor

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
import scala.collection.Map

case class RulesConf(
    rules: Seq[RuleConf]
)

object RulesConf {
  type TagMap = Map[String, Boolean]
  type SigmaMap = Map[String, TagMap]

  def load(spec: String) = {
    val objectMapper = new ObjectMapper(new YAMLFactory())
    objectMapper.findAndRegisterModules()
    objectMapper.readValue(spec, classOf[RulesConf])
  }
}

case class RuleConf(
    rulename: String,
    description: String,
    action: String,
    tags: Seq[TagConf],
    groupby: Seq[String],
    ordered: Boolean,
    parent: String,
    child: String
)

case class TagConf(
    name: String
)
